### DISCUSSION TITLE
Disable for certain projects - Visual Studio

### DISCUSSION THREAD

**Discussion by stefan-marth** [community/community#133485] [Copilot Conversations]
### Select Topic Area

General

### Body

I recently updated Visual Studio Pro and CoPilot is now built into the IDE, vs before where the extensions were able to be used by pressing the icon in the bottom left. Here I could disable copilot based on the solution. Is there a way to do this with the latest version of Visual Studio?

**Comment by danialtavakoli** [comment]
To disable GitHub Copilot in the latest Visual Studio for a specific solution, open your solution, go to "Tools" > "Options" > "GitHub Copilot," and disable it there. If no per-solution option is available, you may need to toggle it globally as needed.

**Comment by ayim** [comment]
@stmarth hello from Visual Studio team - were you using this feature because Copilot was unhelpful/distracting for certain projects, or were you trying to be extra secure and avoid sending certain content for remote inferencing?

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio

**Reply by stefan-marth** [reply]
I’m using it in a setting where specific projects have to be protected, so more for security than anything. Previously, I could disable it on a per-solution basis which was great. Now it seems that functionality is gone and I have to do it globally on a language basis.

**Reply by ayim** [reply]
I see, we've migrated to adopting the GitHub Copilot wide content exclusion functionality for security/protection.
Any chance you're using GitHub Copilot for Business?

https://docs.github.com/en/copilot/managing-copilot/managing-github-copilot-in-your-organization/setting-policies-for-copilot-in-your-organization/about-content-exclusions-for-github-copilot

**Reply by stefan-marth** [reply]
I’m not. My organization isn’t paying for the business subscription. So the personal projects I’m working on is fine for copilot, just when it comes to business apps, I have to disable it.

**Reply by ayim** [reply]
I see. If you're just trying to disable the grey-text code completions, you can bind a shortcut to this command.
It will disable Copilot's grey-text
![image](https://github.com/user-attachments/assets/e288d89d-0b10-4c3f-bb72-535bae106948)

**Reply by stefan-marth** [reply]
Won’t that also disable the intellicode helper as well? I’m looking at only disabling the copilot activity.

**Comment by tshareef4563** [comment]
To disable a project in Visual Studio, right-click the [ kinemaster mod apk ](https://apkblings.com/kinemaster-mod-apk/) project in Solution Explorer and select "Unload Project."

**Comment by ayim** [comment]
It does in fact disable the IntelliCode grey text.
Let me check with developer to see if the per solution command is still available.

-A
